cli/src/main/java/com/devonfw/tools/ide/io/IniFileImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.devonfw.tools.ide.io;
+
+import java.util.LinkedHashMap;
+
+/**
+ * Implementation of {@link IniFile} preserves order of sections and properties between reading and writing
+ */
+public class IniFileImpl implements IniFile {
+
+  LinkedHashMap<String, IniSection> iniMap;
+
+  /**
+   * creates empty IniFileImpl
+   */
+  public IniFileImpl() {
+    this.iniMap = new LinkedHashMap<>();
+  }
+
+  @Override
+  public String[] getSectionNames() {
+    return iniMap.keySet().toArray(new String[0]);
```

Comment:
Same thing but more elegant (since it does not create an empty array that is then discarded but only used to determine the type of the array.
```suggestion
    return iniMap.keySet().toArray(String[]::new);
```